### PR TITLE
librustc_middle: Rename upvar_list to closure_captures

### PR DESCRIPTION
**Pull request by null-sleep**
As part of supporting RFC 2229, we will be capturing all the places that
are mentioned in a closure. Currently the `upvar_list` field gives access to a `FxIndexMap<HirId, Upvar>` map. Eventually this will change, with the `upvar_list` having a more general structure that expresses captured paths, not just the mentioned `upvars`. We will make those changes in subsequent PRs.

This commit modifies the name of the `upvar_list` map to `closure_captures` in `TypeckTables`.

### CODE REVIEW

**Comment by null-sleep** [comment]
@sexxi-goose/rustaceans Tests pass, requesting review

Is the pr message too much? https://github.com/rust-lang/rust/pull/72544 has a much shorter message @arora-aman @ChrisPardy

File: src/librustc_typeck/check/upvar.rs [hunk 1]
```diff
@@ -141,7 +141,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             // This gives us an easier access to them without having to call
             // tcx.upvars again..
             if !upvar_list.is_empty() {
-                self.tables.borrow_mut().upvar_list.insert(closure_def_id, upvar_list);
+                self.tables.borrow_mut().closure_captures.insert(closure_def_id, upvar_list);
```
Comment (arora-aman) [inline src/librustc_typeck/check/upvar.rs]: I think the variable `upvar_list` should be renamed to `closure_captures`

Comment (arora-aman) [reply src/librustc_typeck/check/upvar.rs]: From what I understand this is the type-checking phase, that is where we decide what is the exact type/ places being captured.

Comment (arora-aman) [reply src/librustc_typeck/check/upvar.rs]: From the 2229 plan doc:

> “captures” or “captured paths” will refer to the paths captured by the closure (e.g., x.foo). This list is produced by the type check.

**Comment by arora-aman** [comment]
Also converted it to draft so that we don't mistakenly merge it into master.

**Review by arora-aman — APPROVED** [review]
LGTM

File: src/librustc_typeck/check/upvar.rs (hunk 1 above)
Comment (null-sleep) [reply src/librustc_typeck/check/upvar.rs]: Renamed. Running tests again, they should pass

**Comment by null-sleep** [comment]
> As part of supporting RFC 2229, we will be capturing all the places that
> are mentioned in a closure. Currently the `upvar_list` field gives access to a `FxIndexMap<HirId, Upvar>` map. Eventually this will change, with the `upvar_list` having a more general structure that expresses captured paths, not just the mentioned `upvars`. We will make those changes in subsequent PRs.
> 
> This commit modifies the name of the `upvar_list` map to `closure_captures` in `TypeckTables`.

@arora-aman this is fine?

**Comment by arora-aman** [comment]
I think that is good. Since this is an open source project the more information we can provide the better, reducing the review-loop time is important.

**Comment by arora-aman** [comment]
@jenniferwills btw this will affect the work you've done for using `typechk_table` to read the captures list.

**Comment by jenniferwills** [comment]
@arora-aman nope all good there

**Comment by roxelo** [comment]
Is it just me or this PR also includes commits not related to this change? Kind of makes it hard to review...
Is there a way we can avoid this in the future?

**Comment by roxelo** [comment]
LGTM

**Comment by arora-aman** [comment]
> Is it just me or this PR also includes commits not related to this change? Kind of makes it hard to review...
> Is there a way we can avoid this in the future?

I think it got messed up with the master sync testing, it was fine before that.

**Review by jenniferwills — APPROVED** [review]
LGTM

**Comment by arora-aman** [comment]
For future reference: https://github.com/rust-lang/rust/pull/72591

**Comment by null-sleep** [comment]
+@sexxi-goose/rustaceans second attempt at ping

**Pull request closed without merging**